### PR TITLE
Basic music player

### PR DESCRIPTION
**Pull request by ambern115**
It plays youtube links!
Its organized better?

### CODE REVIEW

**Comment by ambern115** [comment]
Theres probably still some junk left over from testing stuff that I could still remove.

**Comment by posneg** [comment]
I'll be looking into this PR tomorrow

File: bot/music_player/player.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    async def from_url(cls, url, *, loop=None, stream=False):
+        loop = loop or asyncio.get_event_loop()
+        data = await loop.run_in_executor(None, lambda: ytdl.extract_info(url, download=not stream))
+
+        if 'entries' in data:
+            # take first item from a playlist
+            data = data['entries'][0]
+
+        filename = data['url'] if stream else ytdl.prepare_filename(data)
+        #return cls(discord.FFmpegPCMAudio(filename, **ffmpeg_options), data=data)
```
Comment (posneg) [inline bot/music_player/player.py]: Let's remove the extra line here

**Review by posneg — CHANGES_REQUESTED** [review]
Most of the changes look good -- I think really the thing I'd like to see the most is logging, as that'll help debug any issues that arise later.

You can see some existing examples of logging in some of the code I've written, and you can use the bot's logger or you can create a music player specific log/handler.

File: bot/music_player/player_cog.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+
+    @commands.command(
+        brief="Plays a single song",
+        help="""Plays a single song given an mp3's name. If a song is already playing,
+        it adds the song to a queue to be played when the current song ends. Will
+        automatically add bot to a voice channel if its not in one already.""",
+        name="play_local"
+    )
+    async def play_local(self, ctx, song_name, filter=""):
```
Comment (posneg) [inline bot/music_player/player_cog.py:59]: If you want to use multiword song names without quotation marks, you can remove the filter (perhaps make the filter a command per-guild?) argument and use the following function definition instead.
```suggestion
    async def play_local(self, ctx, *, song_name):
```

Comment (posneg) [reply bot/music_player/player_cog.py:59]: After discussion, this sort of change would be in a separate PR.  We'll leave that for later and accept as-is.

**Review by posneg — APPROVED** [review]
LGTM for now

**Pull request merged**